src/widget/dialog.js

```diff
@@ -325,8 +314,16 @@ class Panel {
 
   publicPromise() {
     if (!this.promise) {
-      const then = this.dfd.then.bind(this.dfd)
-      const always = this.dfd.finally.bind(this.dfd)
+      const promise = this.dfd.then(files => {
+        if (this.settings.multiple) {
+          return { v: FileGroup(files, this.settings) }
```

Comment:
Instead of `v` may be just a little better `val` or something like that? Because `val` doesn't reflect sense

Reply:
or `obj`

Reply:
Key name doesn't have sense, because this is hacky way to stop promise chain
I think that will be better if I write comment about that

Reply:
@bautrukevich what do you think?

Reply:
I renamed key and added comment, at the end